internal/attestation/vtpm/attestation.go

```diff
@@ -208,11 +213,13 @@ func (v *Validator) Validate(ctx context.Context, attDocRaw []byte, nonce []byte
 		return nil, fmt.Errorf("validating attestation public key: %w", err)
 	}
 
+	tpmNonce := makeTpmNonce(attDoc.InstanceInfo, extraData)
+
 	// Verify the TPM attestation
 	state, err := tpmServer.VerifyAttestation(
 		attDoc.Attestation,
 		tpmServer.VerifyOpts{
-			Nonce:      extraData,
+			Nonce:      tpmNonce[:],
```

Comment:
Is this ok regarding backwards compatibility (e.g., during an upgrade)?

Reply:
This is a very good point. It'll probably break. I'll need to test it though, but I couldn't image a reason why it wouldn't break

Reply:
Why would it break?
We update software components (the ones who validate attestation statements) first.
Blocking VMs running old software from joining the cluster at this point is intended.
New VMs are running new software (which should have this change) and they supposed to be admitted to the cluster.

Reply:
Okay, Daniel just cleared up my misunderstandings: I thought the (new) CLI would talk aTLS to the (old) cluster during the upgrade, which would not work since the validation logic changed. Since cluster components are upgraded in the order Daniel mentioned, the mechanics in the cluster should continue as expected. But anyhow, I'm currently going through update tests to validate that behavior for existing attestation variants.